src/libraries/Microsoft.Extensions.Configuration.Xml/src/XmlConfigurationElementTextContent.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace Microsoft.Extensions.Configuration.Xml
+{
+    internal class XmlConfigurationElementTextContent
+    {
+        public XmlConfigurationElementTextContent(string textContent, string lineInfo)
+        {
+            TextContent = textContent ?? throw new ArgumentNullException(nameof(textContent));
+            LineInfo = lineInfo ?? throw new ArgumentNullException(nameof(lineInfo));
+        }
+
+        public string TextContent { get; }
+
+        public string LineInfo { get; }
```

Comment:
Do we need to store `LineInfo`? I don't see it used other than for exceptions and it is passed directly from `GetLineInfo(reader)` call as opposed from here.

Reply:
I think we need it. In XmlStreamConfigurationProvider#306 the line info is passed to AddToConfiguration. In that method, a duplicate key exception can be thrown which needs the line info. 

At this point in time, the XML reader has finished its work completely, so we can no longer retrieve the line info from the reader.

Reply:
Yup I missed that part. Disregard this comment.